Fix/remove import unused code

zaerl opened the pull request:
#### Changes proposed in this Pull Request

There are 3 signup steps that are obsolete and must be deleted.

* Remove `/client/signup/steps/import-preview/`
* Remove `/client/signup/steps/import-url/`
* Remove `/client/signup/steps/import-url-onboarding/`
* Remove the entire `import-onboarding` flow

`/start/import-onboarding/site-type` is also not used anymore as well as other obsolete "blue" pages, see: **p58i-bxw-p2**. It is **deliberately not canceled** because we will proceed gradually step by step.

On `client/signup/config/step-components.js` there's a comment regarding a [temporary naming](https://github.com/Automattic/wp-calypso/blob/bc1444b311abe1fe2602a9d332e0c976e8331111/client/signup/config/step-components.js#L19-L21) of:

`from-url` > `import-url` > `import-url-onboarding`

a route that will be removed by this **PR**. These routes are used by the `site-type` page. This step will be removed where the `site-type` and others will be deleted (like `domains` and `plans-import`).

#### Testing instructions

Ensure that these 3 endpoints do not exist anymore.

zaerl committed:
Remove import-preview signup step

zaerl committed:
Remove import-url signup step

zaerl committed:
Remove import-url-onboarding signup step

simison commented:
You can see from Tracks events if any traffic goes through these steps. (`calypso_signup_step_start_X` with `flow` prop "importing" or something like that).

Does the [Wix marketing page](https://wordpress.com/compare/wix/) flow use some of these?

vishnugopal commented on client/signup/config/flows-pure.js:
```diff
@@ -344,11 +344,11 @@ export function generateFlows( {
 		// when switching from the onboarding flow.
 		{
 			name: 'import-onboarding',
```
Why can't we remove this entire flow and all related code? Are we using it somewhere else?

zaerl replied:
I've checked. It can be removed entirely. It is not used anywhere.

vishnugopal commented:
The route does not seem to be used at all: https://mc.a8c.com/tracks/funnel/view.php?32045/2021-11-06/to/2021-12-05

zaerl committed:
Remove import-onboarding signup step

vishnugopal commented on client/signup/config/step-components.js:
```diff
@@ -17,7 +17,6 @@ const stepNameToModuleName = {
 	'domains-theme-preselected': 'domains',
 	'domains-launch': 'domains',
 	'from-url': 'import-url',
```
Should we remove `from-url` and `import-url` here as well?

zaerl replied:
@vishnugopal so we can remove also all: https://github.com/Automattic/wp-calypso/blob/54c36596cc7107b2c4e570505a128d80c2250947/client/signup/config/flows-pure.js#L334-L342

vishnugopal replied:
Well I think `user` and `domains` are used elsewhere, I'm not entirely sure about `plans-import`. We should verify before we remove.

vishnugopal replied:
(It's ok to leave this for a future PR if that makes more sense! I'll test this more thoroughly once you've marked this Ready for Review)

zaerl replied:
@vishnugopal `from-url` is a blue page when you enter a Wix URL, it can be deleted. `domains` is also a blue page (as well as `plans-import`) but I think that it's not a good idea to remove it in order to avoid a potential domino effect. Maybe we just remove the parent `import` and leave the other child routes to the future PR which will delete all other remaining blue pages.

Those pages are outside the scope of this particular PR and AFAIK of the import section.

vishnugopal replied:
Ack, makes sense @zaerl!

vishnugopal commented on client/signup/config/flows-pure.js:
```diff
@@ -340,17 +340,6 @@ export function generateFlows( {
-		{
-			name: 'import-onboarding',
-			steps: [ 'user', 'site-type', 'import-url', 'import-preview', 'domains', 'plans-import' ],
-			destination: getImportDestination,
```
Did we remove references to `getImportDestination` ..? Is that used anywhere else?

zaerl replied:
It's used on another `import` route. This is all legacy code related to the old "blue" pages that can be removed once we remove everything.

zaerl committed:
Remove 'import' flow route

This route contains other routes that will be deleted in a future PR along with all other "blue" obsolete pages.
See: https://github.com/Automattic/wp-calypso/pull/58750#discussion_r763830549

zaerl committed:
Fix: remove unused code

getImportDestination is not used anymore and so it can be removed

vishnugopal reviewed (approved):
This looks good to me! I can't find any regressions or pages that are changed beyond the ones removed. I would request @bogiii to give a second review since we are removing code here!

bogiii commented:
From my side, there are no regressions as well. I think that we are good to go!

cc @zaerl @vishnugopal
